src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -609,6 +612,12 @@ public static Double stringLength (Object o) {
         return (double) s.length();
     }
 
+    public static String normalizeSpace (Object o) {
+        String s = toString(o);
+        String normalized = s.trim().replaceAll("\\s+", " ");
```

Comment:
Java's `\s` includes form feed and vertical tab which are not part of [XML's whitespace definition](https://www.w3.org/TR/REC-xml/#NT-S) but those aren't use anyway so I don't think it matters.

Reply:
@tiritea I think between this note and yours, that's plenty of info about the decision made here.